File: Backend/Models/Donation.js
const mongoose = require('mongoose');

const donationSchema = new mongoose.Schema({
  title: { type: String, required: true },
  targetAmount: { type: Number, required: true },
  currentAmount: { type: Number, default: 0 },
  creatorName: { type: String, required: true },
  creatorId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  isActive: { type: Boolean, default: true } // Field baru untuk status donasi
}, { timestamps: true });

module.exports = mongoose.models.Donation || mongoose.model('Donation', donationSchema);

File: FrontEnd/app/src/main/java/com/example/projectmdp/api/Donation.kt
package com.example.projectmdp.api

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Data class yang merepresentasikan satu entitas Donasi.
 * Strukturnya harus cocok persis dengan JSON yang dikirim oleh server.
 */
@Serializable
data class Donation(
    // Menggunakan @SerialName untuk memetakan field "_id" dari JSON
    // ke properti "id" di Kotlin.
    @SerialName("_id") val id: String,

    val title: String,
    val targetAmount: Long,
    val currentAmount: Long,
    val creatorName: String,
    val creatorId: String,

    // ==========================================================
    // =         PROPERTI YANG DIPERBAIKI (DITAMBAHKAN)         =
    // ==========================================================
    val isActive: Boolean,

    val createdAt: String
)